Automattic/jetpack
# Fixing unchecked references to the JETPACK__PLUGIN_DIR constant

Fixes #

#### Changes proposed in this Pull Request:

* This should avoid fatal errors in cases when the JETPACK__PLUGIN_DIR constant is not defined.
* Check if the jetpack_protect_get_ip() function exists before using it.
* If the function does not exist, check if the JETPACK__PLUGIN_DIR constant is defined.
* If the constant is defined, require the shared-functions.php file, then call the jetpack_protect_get_ip() function.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Open wp shell for a Jetpack site. 
* `yarn docker:wp shell --url=DOMAIN`
* Verify the following wp shell commands return the proper data instead of throwing errors.
* return Automattic\Jetpack\Sync\Modules::get_module( 'options' );
* return Automattic\Jetpack\Sync\Functions::get_modules();

* Go to /projects/plugins/sync-tester
* `composer install`
* Login to your site's wp-admin.
* Check error-log for debug messages.


#### Proposed changelog entry for your changes:


* Sync: Removing references to the JETPACK__PLUGIN_DIR constant.
